library/HTTP/src/Juvix/HTTP/Routes.hs

```diff
@@ -0,0 +1,72 @@
+{-# LANGUAGE DataKinds, FlexibleContexts, FlexibleInstances,
+             MultiParamTypeClasses, TypeFamilies, TypeOperators #-}
+
+
+module Juvix.HTTP.Routes where
+
+import Juvix.Library
+import qualified Juvix.Library.NameSymbol as NameSymbol
+import Servant
+import qualified Juvix.Pipeline
+import qualified Juvix.Backends.Plonk as Plonk
+import qualified Juvix.ToCore.FromFrontend as FF
+import qualified Juvix.Frontend.Types as Frontend
+import qualified Juvix.Sexp as Sexp
+import qualified Juvix.Context as Context
+import qualified Juvix.Core.Base as Core
+import qualified Juvix.Core.HR as HR
+import qualified Juvix.Core.IR as IR
+import Data.Curve.Weierstrass.BLS12381 (Fr)
+
+type ToML = Description "Convert Juvix source code to AST"
+         :> ReqBody '[PlainText] Text
+         :> Post '[JSON] (NameSymbol.T, [Frontend.TopLevel])
+
+type ToSexp = Description "Convert AST to S-expression"
+         :> ReqBody '[JSON] (NameSymbol.T, [Frontend.TopLevel])
+         :> Post '[JSON] (Context.T Sexp.T Sexp.T Sexp.T)
+
+type ToHR = Description "Convert S-expression to HR"
+         :> ReqBody '[JSON] (Context.T Sexp.T Sexp.T Sexp.T)
+         :> Post '[JSON] (FF.CoreDefs HR.T (Plonk.PrimTy Fr) (Plonk.PrimVal Fr))
+
+type ToIR = Description "Convert HR to IR"
+         :> ReqBody '[JSON] (FF.CoreDefs HR.T (Plonk.PrimTy Fr) (Plonk.PrimVal Fr))
+         :> Post '[JSON] (Core.PatternMap Core.GlobalName, FF.CoreDefs IR.T (Plonk.PrimTy Fr) (Plonk.PrimVal Fr))
+
+type ToErased = Description "Convert IR to Erased"
+         :> ReqBody '[JSON] (Core.PatternMap Core.GlobalName, FF.CoreDefs IR.T (Plonk.PrimTy Fr) (Plonk.PrimVal Fr))
+         :> Post '[JSON] (Core.PatternMap Core.GlobalName, FF.CoreDefs IR.T (Plonk.PrimTy Fr) (Plonk.PrimVal Fr))
+
+-- TODO: Abstract to any backend
+type ToCircuit = Description "Convert Erased to Circuit"
+         :> ReqBody '[JSON] (Core.PatternMap Core.GlobalName, FF.CoreDefs IR.T (Plonk.PrimTy Fr) (Plonk.PrimVal Fr))
+         :> Post '[JSON] (Text, Plonk.ArithCircuit Fr) 
+
+type Pipeline = "pipeline" :> 
+    ( ("to-ml" :> ToML) :<|>
+      ("to-sexp" :> ToSexp ) :<|>
+      ("to-hr" :> ToHR ) :<|>
+      ("to-ir" :> ToIR ) :<|>
+      ("to-erased" :> ToErased ) :<|>
+      ("to-circuit" :> ToCircuit ) 
+    )
+
+pipeline :: Server Pipeline
+pipeline = toML :<|>
+           toSexp :<|>
+           toHR :<|>
+           toIR :<|>
+           toErased :<|>
+           toCircuit
+    where
+        toML = notImplemented
+        toSexp = notImplemented
+        toHR = notImplemented
+        toIR = notImplemented
+        toErased = notImplemented
+        toCircuit = notImplemented
+
+
```

Comment:
Does this do dynamic type checking to properly be able to determine which alternative to call?